packages/plugins/plugin-mixpanel/src/methods/group.ts

```diff
@@ -0,0 +1,28 @@
+import type { Mixpanel } from 'mixpanel-react-native';
+import type {
+  GroupEventType,
+  SegmentMixpanelSettings,
+} from '@segment/analytics-react-native';
+
+export default (
+  event: GroupEventType,
+  mixpanel: Mixpanel,
+  settings: SegmentMixpanelSettings
+) => {
+  const groupId = event.groupId;
+  const groupTraits = settings.groupIdentifierTraits;
+
+  if (groupId !== null && groupTraits?.length) {
+    for (let groupTrait of groupTraits) {
+      for (let eventTrait in event.traits) {
+        if (groupTrait.toLocaleLowerCase() === eventTrait.toLocaleLowerCase()) {
+          let group = event.traits[groupTrait] as string;
+          let traits = event.traits;
+
+          mixpanel.getGroup(group, groupId).setOnce('properties', traits);
+        }
+      }
```

Comment:
Per our conversation, unless we know that the `groupTraits` and `eventTraits` casing might be mixed between them you could same some time and code processing this by just accessing directly by key.

```
for (const groupTrait of groupTraits) {
  const group = event.traits[groupTrait] as string;
  const traits = event.traits;

  mixpanel.getGroup(group, groupId).setOnce('properties', traits);
}
```

in another note, you can declare everything with `const` instead of `let` since nothing is changing.